### DISCUSSION TITLE
Would you be open to replacing Cypress with Playwright in official stacks?

### DISCUSSION THREAD

**Discussion by silvenon** [remix-run/remix#3379] [Proposals]
I volunteer to replace Cypress with Playwright in official Remix stacks, would you be open to this? Seeing that you're using Playwright in Remix core.

In short: Playwright makes me want to write more tests. I outlined some of the reasons here: https://github.com/remix-run/remix/discussions/3379#discussioncomment-2884699

**Comment by herecydev** [comment]
Is there something wrong with cypress? Seems very well loved in the community and used far more widely [according to this](https://2021.stateofjs.com/en-US/libraries/testing)

**Reply by sergiodxa** [reply]
Agree, Playwright is a great tool but so is Cypress, and you cold create alternative versions of those stacks using Playwright instead if you want.

**Reply by silvenon** [reply]
Ah, sorry, I forgot to explain why. The primary reason is that Playwright is muuuuch faster, which I assume is why Remix core switched to it too. For Cypress I know I'll always have to wait a while until I find out whether my tests pass, even headlessly, but with Playwright it's almost instant, which really makes E2E testing fun because iterations are fast. 🥰 The second reason is the documentation, it takes time to find references to certain Cypress API, even simple ones like `should()` (I'm still not sure what it can and can't do), making testing more frustrating because it's harder to find straightforward answers. There are other smaller reasons, like the async/await syntax being more developer-friendly, being able to easily change the browser per test, easy dark mode etc.

As for the usage statistic, I assume that's because Cypress has been a great solution for a long time (and still is), and Playwright is new. Compare that to Jest (73%) vs. Vitest (3%), yet the stacks switched to Vitest because it's faster 🤷‍♂️ Although maybe that wouldn't happen if Vitest's API wasn't compatible with Jest, what do you think?

Also, I understand that I can make my own stacks with Playwright, my offer and question was whether there was interest in adding it in the official ones. 🙂

**Reply by fullstackwebdev** [reply]
I switched to playwright from cypress after I got stuck on confusing and or undocumented cypress jwt vs cookie session API handling. I just wanted the puppeteer way to inject auth, then found playwright. To be honest I wouldn't have switched but I couldn't figure it out (in a timely matter) without ejecting to the chome driver.
